tests/vmi_multiqueue_test.go

```diff
@@ -60,26 +59,30 @@ var _ = Describe("[sig-compute]MultiQueue", func() {
 			availableCPUs = tests.GetHighestCPUNumberAmongNodes(virtClient)
 		})
 
-		It("[test_id:4599]should be able to successfully boot fedora to the login prompt with networking mutiqueues enabled without being blocked by selinux", func() {
-			vmi := tests.NewRandomFedoraVMIWithGuestAgent()
-			numCpus := 3
-			Expect(numCpus).To(BeNumerically("<=", availableCPUs),
-				fmt.Sprintf("Testing environment only has nodes with %d CPUs available, but required are %d CPUs", availableCPUs, numCpus),
-			)
-			cpuReq := resource.MustParse(fmt.Sprintf("%d", numCpus))
-			vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceCPU] = cpuReq
-			multiQueue := true
-			vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue = &multiQueue
-			vmi.Spec.Domain.Devices.Rng = &v1.Rng{}
-
-			By("Creating and starting the VMI")
-			vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
-			Expect(err).ToNot(HaveOccurred())
-			tests.WaitForSuccessfulVMIStartWithTimeout(vmi, 360)
-
-			By("Checking if we can login")
-			Expect(libnet.WithIPv6(console.LoginToFedora)(vmi)).To(Succeed())
-		})
+		DescribeTable("should be able to successfully boot fedora to the login prompt with networking mutiqueues enabled without being blocked by selinux",
+			func(createVMISpec func() *v1.VirtualMachineInstance) {
+				vmi := createVMISpec()
+				numCpus := 3
+				Expect(numCpus).To(BeNumerically("<=", availableCPUs),
+					fmt.Sprintf("Testing environment only has nodes with %d CPUs available, but required are %d CPUs", availableCPUs, numCpus),
+				)
+				cpuReq := resource.MustParse(fmt.Sprintf("%d", numCpus))
+				vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceCPU] = cpuReq
+				multiQueue := true
+				vmi.Spec.Domain.Devices.NetworkInterfaceMultiQueue = &multiQueue
+				vmi.Spec.Domain.Devices.Rng = &v1.Rng{}
+
+				By("Creating and starting the VMI")
+				vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+				Expect(err).ToNot(HaveOccurred())
+				tests.WaitForSuccessfulVMIStartWithTimeout(vmi, 360)
+
+				By("Checking if we can login")
+				Expect(console.LoginToFedora(vmi)).To(Succeed())
+			},
+			Entry("[test_id:4599]VM with masquerade binding", tests.NewRandomFedoraVMIWithGuestAgent),
+			Entry("VM with autoattached interfaces", newFedoraVMIWithAutoattachedInterfaces),
```

Comment:
If there is a need to add info to a function is the other one, not this one. `libvmi.NewFedora()` is clean in the sense it limits the hidden additions. You get (almost) a clean VMI.
On the other hand, `NewRandomFedoraVMIWithGuestAgent` adds many things you do not expect.

Perhaps just explicitly mention in the other entry text that a network is included. I have no better idea, but seeing this extra hop is not that nice.

Reply:
Had to do the extra hop so both will have the same function prototype (argument list is different between them)
I can change that the functions will be a factory that on the first do nothing, and on the 2nd removes the interfaces instead.
Will try to push a change like that.

> Perhaps just explicitly mention in the other entry text that a network is included. I have no better idea, but seeing this extra hop is not that nice.

On the first one btw, I used the same method that this test used already
`NewRandomFedoraVMIWithGuestAgent` which is a duplicate of `NewRandomFedoraVMI` but this is other story.
We don't specify nowhere in the function name that we have Masquerade, it is the tests defaults, and used all over.